lib/pbench/agent/tool_meister.py

```diff
@@ -446,29 +514,15 @@ def start(self, env=None):
                 pp,
                 self.args,
             )
-            self.process = subprocess.Popen(
-                " ".join(self.args),
-                cwd=self.tool_dir,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT,
-                env=self.env,
-                shell=True,
-            )
         else:
             self.logger.debug(
                 "Starting persistent tool %s, args %r", self.name, self.args
             )
```

Comment:
Consider collapsing the two branches of this `if` statement into one:
```
            self.logger.debug(
                "Starting persistent tool %s%s, args %r",
                self.name,
                f", env PYTHONPATH={env["PYTHONPATH"]}" if env else "",
                self.args,
            )
```
(I sure hope `PYTHONPATH` actually _is_ in the environment....)

Reply:
And/or, perhaps, move the log call inside `_create_process_with_logger()`?

Reply:
See PR #2697.